packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -112,6 +130,79 @@ class AttachCommand extends FlutterCommand {
     }
   }
 
+
+  Future<int> mdnsQueryDartObservatoryPort() async {
```

Comment:
Lets move this to a separate class?

Reply:
I'm not really sure how much this would benefit from being in a separate class, though IO'm not opposed to it.  Do you have something more specific in mind?

Reply:
Well, the follow up question is can we test it, there is definitely enough logic that It is worth a unit test with mocks. Though I agree we should have an integration test as well.

Rough sketch:

```
class FindAPortAndBetterName {
  FindAPortAndBetterName(this.mdnsClient, this.appId); // Injected for testability

  Future<int> get port;
}
```

Reply:
Done.  Working on tests for it now.  Really glad you pointed this out, in the course of working on it I realized this would break IDEs as is.